src/interpreter/initialization.rs

```diff
@@ -1,20 +1,28 @@
-use super::Interpreter;
+use super::{CheckedMetadata, ExecutableTransaction, InitialBalances, Interpreter, RuntimeBalances};
 use crate::consts::*;
 use crate::context::Context;
 use crate::error::InterpreterError;
-use crate::interpreter::RuntimeBalances;
 use crate::storage::InterpreterStorage;
 
-use fuel_tx::CheckedTransaction;
-use fuel_types::bytes::SizedBytes;
+use fuel_tx::{Checked, IntoChecked, Stage, Transaction};
+use fuel_types::bytes::{SerializableVec, SizedBytes};
 use fuel_types::Word;
 
 use std::io;
 
-impl<S> Interpreter<S> {
+impl<S, Tx> Interpreter<S, Tx>
+where
+    Tx: ExecutableTransaction,
+{
     /// Initialize the VM with a given transaction
-    fn _init(&mut self, tx: CheckedTransaction) -> Result<(), InterpreterError> {
-        self.tx = tx;
+    fn _init(&mut self, tx: Tx, initial_balances: InitialBalances) -> Result<(), InterpreterError> {
+        // TODO: Remove cloning of the transaction and convert it into the `Transaction`.
+        //  VM can work directly with inner types. It is only added to make the
+        //  code as it was before the refactoring.
+        self.tx = tx.clone();
+        let mut transaction: Transaction = tx.into();
```

Comment:
I tried to have the same business logic as before the refactoring.
But it seems we don't need to serialize exactly the `Transaction` into the buffer of the VM. We can serialize `Script` or `Create` without the first 8 bytes that indicate discriminant of the `Transaction` enum.

I will create an issue and follow-up PR if you are okay with this TODO.
Better to do it in a separate PR because it also affects tests and may affect others who try to work with the `memory` field.